VDA-DID-Registry/contracts/IVeridaDIDRegistry.sol

```diff
@@ -23,6 +23,14 @@ interface IVeridaDIDRegistry {
      */
     event SetController(address didAddress, address controller);
 
+    /**
+     * @notice Check whether didAddress is registered
+     * @dev Used in the `VDAXPReward` contract
+     * @param didAddress DID address to be checked
+     * @return bool `true` if registered
+     */
+    function isRegistered(address didAddress) external view returns(bool);
```

Comment:
Any reason you didn't use `lookup()`?

Reply:
This `isRegistered()` function consumes less gas fees than the `lookup()` function.
In some contracts, we need to check whether a DIDAddress is registered in this `DIDRegistry` contrct.  In case, it's convenient to use the `isRegistered()` function instead of `lookup()` function.

Reply:
If it's read only, it doesn't use gas though?

Reply:
It doesn't cosume gas when it's called by the exteranl (through web3).
But if it's called inside a function that uses gases, it consume the gas for this function call too,
In other word, if this `isRegistered()` function is called by the `claimXPReward()` function of the `XPReward` contract, the caller of the `claimXPReward()` function should pay the gas for this `isRegistered()` function too.

Reply:
Ah... that makes sense.